commodore/compile.py

```diff
@@ -136,6 +136,10 @@ def compile(config, cluster_id):
     # Verify that all aliased components support instantiation after resolving component version overrides.
     config.verify_component_aliases(cluster_parameters)
 
+    for cn, component in config.get_components().items():
+        ckey = cn.replace("-", "_")
```

Comment:
If I am not mistaken, this kind of replace is also done elsewhere. As this is a core concept of Commodore, it might be worth to give this its own dedicated function. This might help in future if we need to change how component names are handled.